src/state/user/hooks.tsx

```diff
@@ -401,3 +397,74 @@ export function useTrackedTokenPairs(): [Token, Token][] {
     return Object.keys(keyed).map((key) => keyed[key])
   }, [combinedList])
 }
+
+export function useURLWarningToggle(): () => void {
+  const dispatch = useAppDispatch()
+  return useCallback(() => dispatch(toggleURLWarning()), [dispatch])
+}
+
+export function useOrderValidTo() {
+  const [deadline] = useUserTransactionTTL()
+  return useMemo(() => ({ validTo: calculateValidTo(deadline), deadline }), [deadline])
+}
+
+export function useFavouriteTokens(): Token[] {
+  const { chainId } = useWalletInfo()
+  const serializedTokensMap = useAppSelector(({ user: { favouriteTokens } }) => favouriteTokens)
+
+  return useMemo(() => {
+    if (!chainId) return []
+    const tokenMap: Token[] = serializedTokensMap?.[chainId]
+      ? Object.values(serializedTokensMap[chainId]).map(deserializeToken)
+      : []
+    return tokenMap
+  }, [serializedTokensMap, chainId])
+}
+
+export function useToggleFavouriteToken(): (token: Token) => void {
+  const dispatch = useAppDispatch()
+  return useCallback(
+    (token: Token) => {
+      dispatch(toggleFavouriteToken({ serializedToken: serializeToken(token) }))
+    },
+    [dispatch]
+  )
+}
+
+export function useRemoveAllFavouriteTokens(): () => void {
+  const { chainId } = useWalletInfo()
+  const dispatch = useAppDispatch()
+
+  return useCallback(() => {
+    if (chainId) {
+      dispatch(removeAllFavouriteTokens({ chainId }))
+    }
+  }, [dispatch, chainId])
+}
+
+export function useSelectedWallet(): string | undefined {
+  return useAppSelector(({ user: { selectedWallet } }) => selectedWallet)
+}
+
+export function useInitFavouriteTokens(): void {
+  const { chainId } = useWalletInfo()
+  const dispatch = useAppDispatch()
+
+  return useMemo(() => {
+    if (chainId) {
+      dispatch(initFavouriteTokens({ chainId }))
+    }
+  }, [chainId, dispatch])
+}
+
+export function serializeToken(token: Currency): SerializedToken {
```

Comment:
Repeated fn with the same name and similar implementation.

See src/state/user/reducer

Reply:
The implementation is a little different:
<img width="908" alt="image" src="https://user-images.githubusercontent.com/7122625/233617062-2ba636f7-973a-46a4-92e4-0f5072574470.png">

<img width="555" alt="image" src="https://user-images.githubusercontent.com/7122625/233617085-c32cc28c-636a-4b82-9f1d-27b4f7bf8db6.png">

Reply:
Added TODO, thanks!